### PR TITLE
:seedling: Unit tests badge response

### PR DESCRIPTION
**Pull request by naveensrinivasan**
- Add functions for parsing JSON responses and converting to/from JSON
- Add tests for the badge response functions

#### What kind of change does this PR introduce?

(Is it a bug fix, feature, docs update, something else?)

- [ ] PR title follows the guidelines defined in our [pull request documentation](https://github.com/ossf/scorecard/blob/main/CONTRIBUTING.md#pr-process)

#### What is the new behavior (if this is a feature change)?**

- [ ] Tests for the changes have been added (for bug fixes/features)

#### Does this PR introduce a user-facing change?

For user-facing changes, please add a concise, human-readable release note to
the `release-note`

(In particular, describe what changes users might need to make in their
application as a result of this pull request.)

```release-note

```

### CODE REVIEW

File: clients/cii_response_test.go [hunk 1]
```diff
@@ -0,0 +1,166 @@
...
+				return
+			}
+			if !reflect.DeepEqual(got, tt.want) {
+				t.Errorf("ParseBadgeResponseFromJSON() got = %v, want %v", got, tt.want)
+			}
+		})
+	}
+}
+
+func TestBadgeResponse_AsJSON(t *testing.T) {
```
Comment (spencerschrock) [inline clients/cii_response_test.go:68]: t.parallel?

**Review by spencerschrock — APPROVED** [review]
Thanks.

Optional: The `t.Errorf` calls with a `return` after them could be replaced with a `t.Fatalf`

File: clients/cii_response_test.go (hunk 1 above)
Comment (naveensrinivasan) [reply clients/cii_response_test.go:68]: 👍

**Comment by naveensrinivasan** [comment]
> Thanks.
> 
> Optional: The `t.Errorf` calls with a `return` after them could be replaced with a `t.Fatalf`

👍

**Pull request merged**